### PR TITLE
Move Main and classes it depends on to its own package

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

**Comment by eregon** [comment]
This is potentially going to break many things: renaming the Main's package.
We could make it a separate Mx project instead and packages would not matter, but then the separation would be enforced at the IDE/Java compilation level, which I think is better and prevent introducing a dependency back.

**Comment by eregon** [comment]
Moving to a separate package might help to clarify the dependencies though.

**Comment by pitr-ch** [comment]
yeah, already added a deprecated Main class for compatibility, until we remove its references from GraalVM. Making it a separate package is a next step.

**Comment by pitr-ch** [comment]
Please don't review until further notice. I'll move the launcher to a separate mx project.

**Comment by pitr-ch** [comment]
Ready for review.

File: mx.truffleruby/suite.py [hunk 1]
```diff
@@ -264,11 +277,23 @@
             "distDependencies": [
                 "truffle:TRUFFLE_API",
                 "truffle:TRUFFLE_NFI",
+                "truffleruby:TRUFFLERUBY-LAUNCHER",
```
Comment (eregon) [inline mx.truffleruby/suite.py]: This looks good.
Could you check with `jar tf .../foo.jar | grep Main` that the launcher `.class` files are only in the truffleruby-launcher.jar?

File: src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java [hunk 2]
```diff
@@ -432,7 +432,7 @@ private void processArgument() throws CommandLineException {
                         config.setOption(OptionsCatalog.VERBOSITY, "true");
                         break FOR;
                     } else if (argument.startsWith("--dump=")) {
-                        LogWithoutTruffle.LOGGER.warning("the --dump= switch is silently ignored as it is an internal development tool");
+                        TruffleLessLog.LOGGER.warning("the --dump= switch is silently ignored as it is an internal development tool");
```
Comment (eregon) [inline src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java]: These usages of the Log seem rather simple, we could probably just print to stderr here.
The Logger level itself could be passed as an option to the context.

File: mx.truffleruby/suite.py (hunk 1 above)
Comment (pitr-ch) [inline mx.truffleruby/suite.py]: yeah already checked, works as expected

File: src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java (hunk 2 above)
Comment (pitr-ch) [inline src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java]: yeah I thought about that too, but we would no longer manage all warnings/etc. with logger. So I am in favor of keeping the 2 logger classes.

**Pull request merged**